runtime/domain.c

```diff
@@ -150,13 +150,11 @@ CAMLexport __thread caml_domain_state* Caml_state;
 
 asize_t caml_norm_minor_heap_size (intnat wsize)
 {
-  asize_t page_size = caml_mem_round_up_pages(1);
   asize_t bs, max;
   if (wsize < Minor_heap_min) wsize = Minor_heap_min;
   bs = caml_mem_round_up_pages(Bsize_wsize (wsize));
 
-  Assert(page_size * 2 < Minor_heap_max);
-  max = Minor_heap_max - page_size * 2;
```

Comment:
@kayceesrk I am unsure about the requirement for `- page_size * 2` here.
Is it related to guard pages of some sort that used to be put in place? Or am I missing something?

Reply:
IIRC, this was when the DLS area was maintained just beside the minor heap area. Currently, we maintain the Caml_state in r14 register on amd64. Earlier, this was not maintained in a separate register. It was computed from the young_pointer, which is maintained in the r15 register, by applying a mask. Now that we maintain Caml_state in r14 and Caml_state is anyway maintained separately, we don't need this `page_size * 2`.